openupgrade_scripts/scripts/loyalty/16.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+_field_renames = [
+    ("coupon.program", "coupon_program", "promo_applicability", "applies_on"),
+    ("coupon.program", "coupon_program", "promo_code_usage", "trigger"),
+    ("coupon.program", "coupon_program", "maximun_use_number", "max_usage"),
+]
+_models_renames = [("coupon.program", "loyalty.program")]
+_tables_renames = [("coupon_program", "loyalty_program")]
+
+
+def set_applies_on(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET applies_on =
+            CASE
+                WHEN applies_on = 'on_current_order' THEN 'current'
+                WHEN applies_on = 'on_next_order' THEN 'future'
+            END
+        """,
+    )
+
+
+def set_trigger(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET trigger =
+            CASE
+                WHEN trigger = 'no_code_needed' THEN 'auto'
+                WHEN trigger = 'code_needed' THEN 'with_code'
+            END
+        """,
+    )
+
+
+def set_program_type(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET program_type =
+            CASE
+                WHEN program_type = 'coupon_program' THEN 'coupons'
+                WHEN program_type = 'promotion_program' THEN 'promotion'
+            END
+        """,
+    )
+
+
+def set_limit_usage_values(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET limit_usage = TRUE
+        WHERE max_usage IS NOT NULL;
+        """,
+    )
+
+
+def fill_date_to_field(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE loyalty_program
+        ADD COLUMN IF NOT EXISTS date_to VARCHAR
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET date_to = create_date + validity_duration * interval '1 day'
+        WHERE validity_duration != 0 AND date_to IS NULL;
+        """,
+    )
+
+
+def set_portal_visible_default_values(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE loyalty_program
+        SET portal_visible =
+            CASE
+                WHEN program_type IN ('gift_card', 'ewallet', 'loyalty', 'next_order_coupons') THEN true
+                ELSE false
+            END
```

Comment:
Why this? The default value is `False`

This would be a simpler implementation, btw:
```suggestion
        SET portal_visible = program_type IN ('gift_card', 'ewallet', 'loyalty', 'next_order_coupons')
```

Reply:
In this method the default values are set depending on the type of promotion, in this case I have taken the values of portal_visible to adapt whatever comes.
https://github.com/odoo/odoo/blob/16.0/addons/loyalty/models/loyalty_program.py#L188